pytest_catchlog.py

```diff
@@ -154,6 +175,72 @@ def pytest_runtest_teardown(self, item):
         with self._runtest_for(item, 'teardown'):
             yield
 
+    def configure_console_handler(self, config):
+        verbosity = config.getoption('-v')
+        # Prepare the handled_levels dictionary
+        log_levels = []
+        handled_levels = {}
+        available_levels = set([
+            #logging.CRITICAL,  # This will be the default console level if no -v is passed
+            logging.ERROR,
+            logging.WARNING,
+            logging.INFO,
+            logging.DEBUG,
+            logging.NOTSET
+        ])
+        for level in get_option_ini(config, 'log_extra_levels'):
+            try:
+                level_num = int(getattr(logging, level, level))
+            except ValueError:
+                # Python logging does not recognise this as a logging level
+                raise pytest.UsageError(
+                    '\'{0}\' is not recognized as a logging level name. Please '
+                    'consider passing the logging level num instead.'.format(
+                        level
+                    )
+                )
+            available_levels.add(level_num)
+
+        for level in available_levels:
+            if level > logging.WARN:
+                # -v set's the console handler logging level to ERROR, higher
+                # log level messages, ie, >= FATAL are always shown
+                continue
+            if level < logging.NOTSET:
+                # Log levels lower than NOTSET, we're not interested
+                continue
+            if level in log_levels:
+                # We already know about this log level
+                continue
+            log_levels.append(level)
+
+        # Reverse the list because we're interested on higher logging levels
+        # first
+        log_levels = sorted(log_levels, reverse=True)
```

Comment:
This is a matter of preferences, but it could also be handled by the following one-liner:
```py
log_levels = sorted(set(level for level in available_levels
                        if logging.NOTSET <= level <= logging.WARNING),
                    reverse=True)
```